Realm/Realm/DatabaseTypes/Accessors/ManagedAccessor.cs

```diff
@@ -0,0 +1,268 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Realms.Exceptions;
+using Realms.Extensions;
+using Realms.Schema;
+
+namespace Realms
+{
+    internal class ManagedAccessor
+        : IManagedAccessor
+    {
+        private Lazy<int> _hashCode;
+
+        private Realm _realm;
+
+        private ObjectHandle _objectHandle;
+
+        private Metadata _metadata;
+
+        private NotificationTokenHandle _notificationToken;
+
+        private Action<string> _onNotifyPropertyChanged;
+
+        public ObjectHandle ObjectHandle => _objectHandle;
+
+        public bool IsManaged => true;
+
+        public bool IsValid => _objectHandle?.IsValid != false;
+
+        public bool IsFrozen => _realm?.IsFrozen == true;
+
+        public Realm Realm => _realm;
+
+        public ObjectSchema ObjectSchema => _metadata?.Schema;
+
+        public int? HashCode => _hashCode.Value;
+
+        public int BacklinksCount => _objectHandle?.GetBacklinkCount() ?? 0;
+
+        public IThreadConfinedHandle Handle => _objectHandle;
+
+        public Metadata Metadata => _metadata;
+
+        public RealmObjectBase.Dynamic DynamicApi => new(this);
+
+        private ManagedAccessor(Realm realm,
+            ObjectHandle objectHandle,
+            Metadata metadata)
+        {
+            _realm = realm;
+            _objectHandle = objectHandle;
+            _metadata = metadata;
+            _hashCode = new Lazy<int>(() => _objectHandle.GetObjHash());
+        }
+
+        public static ManagedAccessor Create(Realm realm,
```

Comment:
What's the value of a factory vs calling the constructor directly?

Reply:
Sounds like you've socialized a bit too much with Claus and the rest of the Java team 😅 Let's use a simple constructor.